### PR TITLE
[ADD] DCOS-1551 cluster status

### PR DESCRIPTION
**Pull request by YuraTolochkevich**

### CODE REVIEW

**Comment by swartzrock** [comment]
Hi Yura, looks great! I just had one more fix necessary - please see https://mesosphere.atlassian.net/browse/DCOS-1972.

I want to avoid adding extra configurations unless necessary.

**Comment by mgummelt** [comment]
Why are we calling this `dcos cluster`?  I thought it was going to be `dcos status`

File: cli/dcoscli/cluster/main.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
+"""Provides status about the DCOS cluster managed by this CLI
+
+Usage:
+   dcos cluster --help
```
Comment (mgummelt) [inline cli/dcoscli/cluster/main.py]: This isn't how we do --help, please see the other commands as an example

File: cli/dcoscli/cluster/main.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
+"""Provides status about the DCOS cluster managed by this CLI
```
Comment (mgummelt) [inline cli/dcoscli/cluster/main.py]: "Get the status of the DCOS cluster"

File: cli/dcoscli/cluster/main.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+from dcos import cmds, emitting, util
+from dcos.errors import DCOSException
+from dcos.http import request
+from dcoscli import tables
+
+logger = util.get_logger(__name__)
+
+emitter = emitting.FlatEmitter()
+
+MESOS_STATE_URL = '{}:5050/state.json'
```
Comment (mgummelt) [inline cli/dcoscli/cluster/main.py]: See the functions in `mesos.py` for accessing mesos.  Specifically, the `MasterClient` class.

File: cli/dcoscli/cluster/main.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    """
+    cluster_data, exit_code = get_cluster_data()
+    emitting.publish_table(emitter, cluster_data['Components'],
+                           tables.cluster_table, is_json)
+    if exit_code != 0:
+        raise DCOSException()
+
+
+def _info():
+    """Print task cli information.
```
Comment (mgummelt) [inline cli/dcoscli/cluster/main.py]: Looks like you copied and pasted from task/main.py.  Please update

File: cli/dcoscli/cluster/main.py [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+    :returns: process return code
+    :rtype: int
+    """
+
+    emitter.publish(__doc__.split('\n')[0])
+    return 0
+
+
+def get_cluster_data():
```
Comment (mgummelt) [inline cli/dcoscli/cluster/main.py]: Make sure all private functions are prefixed with a `_`

File: cli/dcoscli/cluster/main.py [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+            hierarchy=['cluster', 'status'],
+            arg_keys=['--json'],
+            function=_status),
+        ]
+
+
+def _status(is_json):
+    """ Print to cli cluster status
+
+    :param: args: dcos cluster input arguments
```
Comment (mgummelt) [inline cli/dcoscli/cluster/main.py]: param is wrong

File: cli/dcoscli/cluster/main.py [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+    type: args: dict
```
Comment (mgummelt) [inline cli/dcoscli/cluster/main.py]: this is the wrong pydoc format

File: cli/dcoscli/cluster/main.py [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+    :rtype: dict
+    """
+    exit_code_list = list()
+
+    def sum_codes(f):
+        status, exit_code = f()
+        exit_code_list.append(exit_code)
+        return status
+
+    return {'Components': [{'Name': 'Mesos Master',
```
Comment (mgummelt) [inline cli/dcoscli/cluster/main.py]: It looks like you only ever use the 'Components' key.  If that's the case, just remove it.  It's unnecessary.

**Comment by YuraTolochkevich** [comment]
@mgummelt Thanks Michael! Please see updates

**Comment by swartzrock** [comment]
I still see the "dcos_url" parameter, required in order to use this. Can you please remove this? Try deleting ~/.dcos/ and rerunning to see the error.

**Comment by mgummelt** [comment]
I still need an answer to my question above regarding naming.

**Comment by YuraTolochkevich** [comment]
@mgummelt, ah, sorry, I forgot to notice, I removed "cluster" word from command"

**Comment by YuraTolochkevich** [comment]
@mgummelt Hello Michael! are there any more comments to this PR?

**Comment by mgummelt** [comment]
I'm talking with @swartzrock about it now, I'll get back to you

**Comment by mgummelt** [comment]
@swartzrock and I talked and agreed that this makes more sense as an installable package than as a core CLI command.  Sorry for the back and forth.  I'm going to close this PR and migrate the code back into a universe or multiverse package.

**Pull request closed without merging**